graylog2-web-interface/src/components/configurations/McpConfig.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import * as React from 'react';
+import { useEffect, useState } from 'react';
+
+import { useStore } from 'stores/connect';
+import type { Store } from 'stores/StoreTypes';
+import { ConfigurationsActions, ConfigurationsStore } from 'stores/configurations/ConfigurationsStore';
+import { getConfig } from 'components/configurations/helpers';
+import { ConfigurationType } from 'components/configurations/ConfigurationTypes';
+import { BootstrapModalForm, Button, Input } from 'components/bootstrap';
+import { IfPermitted } from 'components/common';
+import Spinner from 'components/common/Spinner';
+import 'moment-duration-format';
+
+type McpConfigState = {
+  enable_remote_access: boolean;
+};
+
+const McpConfig = () => {
+  const [showConfigModal, setShowConfigModal] = useState<boolean>(false);
+  const [modalConfig, setModalConfig] = useState<McpConfigState|undefined>(undefined);
+  const [viewConfig, setViewConfig] = useState<McpConfigState|undefined>(undefined);
+  const configuration = useStore(ConfigurationsStore as Store<Record<string, any>>, (state) => state?.configuration);
+
+  useEffect(() => {
+    ConfigurationsActions.list(ConfigurationType.MCP_CONFIG).then(() => {
+      const config = getConfig(ConfigurationType.MCP_CONFIG, configuration);
+      setViewConfig(config);
+      setModalConfig(config);
+    });
+  }, [configuration]);
+
+  const openModal = () => {
+    setShowConfigModal(true);
+  };
+
+  const onModalClickEnableRemoteAccess = () => {
+    setModalConfig({...modalConfig, enable_remote_access: !modalConfig.enable_remote_access});
+  };
+
+  const onModalCancel = () => {
+    setShowConfigModal(false);
+    setModalConfig(viewConfig);
+  };
+
+  const onModalSave = () => {
+    // todo send telemetry
+    ConfigurationsActions.update(ConfigurationType.MCP_CONFIG, {...modalConfig}).then(() => {
+      setShowConfigModal(false);
+    });
+  };
+
+  if (!viewConfig) {
+    return <Spinner />;
+  }
+
+  return (
+    <div>
+      <h2>MCP Configuration</h2>
+
+      <dl className="deflist">
+        <dt>Remote MCP access</dt>
+        <dd>{viewConfig.enable_remote_access ? "Enabled" : "Disabled"}</dd>
+      </dl>
+
+      {/* todo: set proper role here 👇*/}
+      <IfPermitted permissions="clusterconfigentry:edit">
```

Comment:
Is clusterconfigentry:edit the right permission to use here?

Reply:
Yes, unfortunately there's no good way to do more finegrained checks, this is a known problem and we will improve this at a later day. For now, that check is correct.